### ISSUE TITLE
Searching a shared library fails

### ISSUE THREAD

**Issue by tdorsey** [pkkid/python-plexapi#174]
refs #155

I'm trying to access a shared library via myplex. 

```
my_server = my_account.resource(my_server_name).connect()
results = my_server.library.section(section).search() //Works, 


their_server = my_account.resource(their_server_name).connect()
results = their_server.library.section(section).search() //Bad Request, due to empty Library() 

``` 
https://github.com/pkkid/python-plexapi/blob/master/plexapi/server.py#L167 has a note that only the owner has access to the /library endpoint, so when this is called on a shared server, it returns an empty library. 
What's the recommend way to query a shared library when you are not the owner? https://github.com/pkkid/python-plexapi/issues/155#issuecomment-318649702 has a note that /library/sections works, but there's no implementation for it. 
I can submit a PR for this, but wanted to make sure it's not handled elsewhere already.

**Comment by Hellowlol** [comment]
I dont understand. This should already be handled. The point about adding a empty library is so we can still access it. (using search for what ever. ) Did you get any traceback?

**Comment by tdorsey** [comment]
```
Traceback (most recent call last):
  File "site-packages\plexapi\server.py", line 164, in library
    data = self.query(Library.key)
  File "site-packages\plexapi\server.py", line 346, in query
    raise BadRequest('(%s) %s; %s %s' % (response.status_code, codename, response.url, errtext))
plexapi.exceptions.BadRequest: (401) unauthorized; https://<redacted>.<redacted>.plex.direct:32400/library?X-Plex-Token=<redacted> <html><head><title>
Unauthorized</title></head><body><h1>401 Unauthorized</h1></body></html>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".\plexsync.py", line 126, in <module>
    their_media = getMedia(their_server, section)
  File ".\plexsync.py", line 63, in getMedia
    results = server.library.section(section).search()
  File "site-packages\plexapi\server.py", line 169, in library
    return Library(self, {})
  File "site-packages\plexapi\base.py", line 46, in __init__
    self._loadData(data)
  File "site-packages\plexapi\library.py", line 26, in _loadData
    self.identifier = data.attrib.get('identifier')
AttributeError: 'dict' object has no attribute 'attrib'
```

at a casual glance, it's checking the /library endpoint of a server it doesn't own and failing. setting Library.key to 'library/sections' when it's a shared server appears to resolve the issue

**Comment by Hellowlol** [comment]
Yeah. See https://github.com/pkkid/python-plexapi/pull/175

**Comment by tdorsey** [comment]
closed by #175
